File: app/src/main/java/pl/piotrskiba/angularowo/utils/PreferenceUtils.java
package pl.piotrskiba.angularowo.utils;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import pl.piotrskiba.angularowo.R;

public class PreferenceUtils {

    public static String getAccessToken(Context context){
        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
        return sharedPreferences.getString(context.getString(R.string.pref_key_access_token), null);
    }

    public static String getUsername(Context context){
        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
        return sharedPreferences.getString(context.getString(R.string.pref_key_nickname), null);
    }

    public static String getRankName(Context context){
        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
        return sharedPreferences.getString(context.getString(R.string.pref_key_rank), null);
    }
}

File: app/src/main/java/pl/piotrskiba/angularowo/utils/PreferenceUtils.kt
package pl.piotrskiba.angularowo.utils

import android.content.Context
import androidx.preference.PreferenceManager
import pl.piotrskiba.angularowo.R

class PreferenceUtils(private val context: Context) {

    private val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)

    var uuid: String?
        get() = sharedPreferences.getString(context.getString(R.string.pref_key_uuid), null)
        set(value) {
            sharedPreferences.edit().apply {
                putString(context.getString(R.string.pref_key_uuid), value)
                apply()
            }
        }

    var username: String?
        get() = sharedPreferences.getString(context.getString(R.string.pref_key_nickname), null)
        set(value) {
            sharedPreferences.edit().apply {
                putString(context.getString(R.string.pref_key_nickname), value)
                apply()
            }
        }
}